src/databricks/labs/ucx/source_code/notebooks/cells.py

```diff
@@ -496,27 +494,57 @@ def _make_commands_for_magic_command_call_nodes(cls, nodes: list[Call]):
         for node in nodes:
             arg = node.args[0]
             if isinstance(arg, Const):
-                yield MagicCommand(node, arg.value)
+                yield MagicLine(node, arg.value)
 
     def __init__(self, node: NodeNG, command: bytes):
         super().__init__(node)
         self._command = command.decode()
 
-    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+    def as_magic(self) -> MagicCommand | None:
         if self._command.startswith("%pip") or self._command.startswith("!pip"):
-            cmd = PipMagic(self._command)
-            return cmd.build_dependency_graph(graph)
+            return PipCommand(self._command)
+        if self._command.startswith("%run"):
+            return RunCommand(self._command)
+        return None
+
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+        magic = self.as_magic()
+        if magic is not None:
+            return magic.build_dependency_graph(graph)
         problem = DependencyProblem.from_node(
             code='unsupported-magic-line', message=f"magic line '{self._command}' is not supported yet", node=self.node
         )
         return [problem]
 
 
-class PipMagic:
+class MagicCommand(ABC):
 
     def __init__(self, code: str):
         self._code = code
 
+    @abstractmethod
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]: ...
+
+
+class RunCommand(MagicCommand):
+
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+        # TODO
+        return []
+
+    @property
+    def notebook_path(self) -> Path | None:
+        for quote in ('"', "'"):
+            start = self._code.find(quote)
+            if start >= 0:
+                end = self._code.find(quote, start + 1)
+                if end >= 0:
+                    return Path(self._code[start + 1 : end])
+        return None
```

Comment:
The quotes for the path areoptional; they're only required if there is whitespace in the path.

(Most of the solacc examples omit the quotes; people seem to only use them when required.)

I'd really like to see a unit test dedicated to just this property, although updating the samples to exercise the variants would also be fine.

It also looks like `RunCell` has an independent implementation of the same thing in `_read_notebook_path()`.

Reply:
Thanks for that. Fixed the optionality. The reason we have a different implementation is because run cells can span multiple lines whereas magic lines can only be one-liners